front/lib/client/handle_file_upload.ts

```diff
@@ -50,7 +50,7 @@ export async function handleFileUploadToText(
               return item.str;
             }
           });
-          text += strings.join(" ") + "\n";
+          text += `Page: ${pageNum}/${pdf.numPages}\n${strings.join(" ")}\n\n`;
```

Comment:
Note: ideally in the future we will use Sections to upsert these documents as well :+1: